test/modules/EdgeHubRestartTester/src/Program.cs

```diff
@@ -0,0 +1,318 @@
+    using Newtonsoft.Json;
+    using Message = Microsoft.Azure.Devices.Client.Message;
+
+    class Program
+    {
+        static readonly ILogger Logger = ModuleUtil.CreateLogger("EdgeHubRestartTester");
+
+        static uint restartCount = 0;
+        static long messageCount = 0;
+        static long directMethodCount = 0;
+
+        public static int Main() => MainAsync().Result;
+
+        static async Task<int> MainAsync()
+        {
+            Guid batchId = Guid.NewGuid();
+            Logger.LogInformation($"Starting Edge Hub Restart Tester ({batchId}) with the following settings:\r\n{Settings.Current}");
+
+            Logger.LogInformation($"Load gen delay start for {Settings.Current.TestStartDelay}.");
+            await Task.Delay(Settings.Current.TestStartDelay);
+
+            (CancellationTokenSource cts, ManualResetEventSlim completed, Option<object> handler) = ShutdownHandler.Init(TimeSpan.FromSeconds(5), Logger);
+
+            ServiceClient iotHubServiceClient = null;
+            ModuleClient msgModuleClient = null;
+            ModuleClient dmModuleClient = null;
+            try
+            {
+                iotHubServiceClient = ServiceClient.CreateFromConnectionString(Settings.Current.ServiceClientConnectionString);
+
+                if (Settings.Current.MessageEnable)
+                {
+                    msgModuleClient = await ModuleUtil.CreateModuleClientAsync(
+                        Settings.Current.MessageTransportType,
+                        ModuleUtil.DefaultTimeoutErrorDetectionStrategy,
+                        ModuleUtil.DefaultTransientRetryStrategy,
+                        Logger);
+                    msgModuleClient.OperationTimeoutInMilliseconds = Settings.Current.SdkRetryTimeout;
+                }
+
+                if (Settings.Current.DirectMethodEnable)
+                {
+                    dmModuleClient = await ModuleUtil.CreateModuleClientAsync(
+                        Settings.Current.MessageTransportType,
+                        ModuleUtil.DefaultTimeoutErrorDetectionStrategy,
+                        ModuleUtil.DefaultTransientRetryStrategy,
+                        Logger);
+                }
+
+                DateTime testStart = DateTime.UtcNow;
+                DateTime testExpirationTime = testStart + Settings.Current.TestDuration;
+                Dictionary<string, Task<Tuple<DateTime, HttpStatusCode>>> taskList = new Dictionary<string, Task<Tuple<DateTime, HttpStatusCode>>>();
+
+                while ((!cts.IsCancellationRequested) && (DateTime.UtcNow < testExpirationTime))
+                {
+                    DateTime eachTestExpirationTime = testStart.AddMinutes(Settings.Current.RestartIntervalInMins);
+                    (DateTime restartTime, HttpStatusCode restartStatus) = await RestartModules(iotHubServiceClient);
+
+                    // Increment the counter when issue an edgeHub restart
+                    restartCount++;
```

Comment:
why do you need restart count if it is always the same as message count and dm count?

Reply:
Ideally they should be the same, but you never know. 
I'm adding it here as a sanity check.

Reply:
Again I can't think of any case it will be different. If it is different, it is bug in our test code.